clustercache/clustercache.go

```diff
@@ -0,0 +1,240 @@
+	if isInCache {
+		return clusterNameOrID, nil
+	}
+
+	auxParams := clientWrapper.DefaultAuxiliaryParams()
+	auxParams.ActivityName = listClustersActivityName
+
+	// Get a list of all clusters
+	response, err := clientWrapper.GetClusters(auxParams)
+	if err != nil {
+		switch {
+		case clienterror.IsUnauthorizedError(err):
+			return "", microerror.Mask(errors.NotAuthorizedError)
+
+		case clienterror.IsAccessForbiddenError(err):
+			return "", microerror.Mask(errors.AccessForbiddenError)
+
+		default:
+			return "", microerror.Mask(err)
+		}
+	}
+
+	var (
+		// IDs that correspond to the same cluster name
+		matchingIDs []string
+		// All IDs returned from the response
+		allClusterIDs = make([]string, 0, len(response.Payload))
+	)
+	for _, cluster := range response.Payload {
+		allClusterIDs = append(allClusterIDs, cluster.ID)
+		// Check if this is the cluster we're looking for
+		if matchesValidation(clusterNameOrID, cluster) {
+			matchingIDs = append(matchingIDs, cluster.ID)
+		}
+	}
+
+	CacheIDs(endpoint, allClusterIDs)
+
+	if matchingIDs == nil {
+		// There are no IDs that correspond to that cluster name
+		return "", microerror.Mask(errors.ClusterNotFoundError)
+	} else if len(matchingIDs) > 1 {
+		// There are multiple IDs that correspond to that cluster name
+		// Help the user decide which cluster to pick
+		_, id := handleNameCollision(clusterNameOrID, response.Payload)
+
+		return id, nil
+	}
+
+	return matchingIDs[0], nil
+}
+
+// New creates a new Cache object
+func New() *Cache {
+	c := &Cache{}
+	c.Endpoints = Endpoints{}
+
+	return c
+}
+
+// IsInCache checks if a cluster ID is present in the
+// persistent cluster cache
+func IsInCache(endpoint string, ID string) bool {
+	var (
+		err                error
+		endpointExpiration time.Time
+
+		now = time.Now()
+	)
+	// Read the existing cache file
+	existing, err := read(config.FileSystem)
+	if err != nil {
+		return false
+	}
+
+	c := existing.Endpoints[endpoint]
+	for _, cID := range c.IDs {
+		// Check if the cache is expired
+		endpointExpiration, err = time.Parse(timeLayout, c.Expiry)
+		if err != nil || now.After(endpointExpiration) {
+			return false
+		}
+		// This is the one we're looking for
+		if cID == ID {
+			return true
+		}
+	}
+
+	return false
+}
+
+// CacheIDs adds cluster IDs to a persistent cache,
+// which can be used for decreasing timeout in getting
+// cluster IDs, for commands that take both cluster names and IDs
+func CacheIDs(endpoint string, c []string) {
+	// Let's not store an empty list
+	if len(c) == 0 {
+		return
+	}
+
+	fs := config.FileSystem
+
+	var cache *Cache
+	{
+		// Create a new Cache object if there is no file there yet
+		cache, _ = read(fs)
+		if cache == nil {
+			cache = New()
+		}
+	}
+
+	// Add the cache to a certain endpoint
+	cache.Endpoints[endpoint] = EndpointCache{
+		Expiry: time.Now().Add(cacheDuration).Format(timeLayout),
+		IDs:    c,
+	}
+
+	// Write the cache file
+	_ = write(fs, cache)
```

Comment:
Same as above: if the return value of `write` is not interesting, then rather make it not return anything.